src/platform/macos/nv12_zero_device.cpp

```diff
@@ -17,23 +17,19 @@ namespace platf {
     av_frame_free(&frame);
   }
 
+  void
+  free_buffer(void *opaque, uint8_t *data) {
+    CVPixelBufferRelease((CVPixelBufferRef) data);
+  }
+
   int
   nv12_zero_device::convert(platf::img_t &img) {
-    av_frame_make_writable(av_frame.get());
-
     av_img_t *av_img = (av_img_t *) &img;
 
-    // Set up the data fields in the AVFrame to point into the mapped CVPixelBuffer
-    int planes = CVPixelBufferGetPlaneCount(av_img->pixel_buffer->buf);
-    for (int i = 0; i < planes; i++) {
-      av_frame->linesize[i] = CVPixelBufferGetBytesPerRowOfPlane(av_img->pixel_buffer->buf, i);
-      av_frame->data[i] = (uint8_t *) CVPixelBufferGetBaseAddressOfPlane(av_img->pixel_buffer->buf, i);
-    }
+    av_buffer_unref(&av_frame->buf[0]);
 
-    // We just set data pointers to point into our CVPixelBuffer above, so we have to hold
-    // a reference to these buffers to keep them around until the AVFrame is done using them.
-    backing_img.sample_buffer = av_img->sample_buffer;
-    backing_img.pixel_buffer = av_img->pixel_buffer;
+    av_frame->buf[0] = av_buffer_create((uint8_t *) CFRetain(av_img->pixel_buffer->buf), 0, free_buffer, NULL, 0);
```

Comment:
Maybe add a couple of words why we use `av_buffer_create()` here, since we don't do it anywhere else.
I understood it as ffmpeg "smart-pointer" that will be freed in `av_frame_free()`.

Reply:
Added a comment here.